tests/api-watch/test_diff_modtype_decl.ml

```diff
@@ -0,0 +1,98 @@
+open Api_watch
+open Test_helpers
+
+let%expect_test "Module types with both value and submodule changes" =
+  let reference =
+    compile_interface
+      {|
+    type ('a, 'b) result = Ok of 'a | Error of 'b
+    val f : int -> string
```

Comment:
Same remark as with the cram tests. This can be cleared out as it is not required to test that we correctly diff modtypes.

Same goes to the other tests below. Having minimal tests is important to clearly exhibit the behaviour under test and to prevent test breakage from happening too often. For instance this test would break if we changed how we represent or display value changes even though that's not what we're testing here.